tests/unit/cli/test_pipeline.py

```diff
@@ -11,3 +11,238 @@
+
+
+class StepConfig(BaseStepConfig):
+    some_option: int = 4
+
+@step
+def {STEP_NAME}(config: StepConfig) -> Output(output_1={CUSTOM_OBJ_NAME},
+                                              output_2=int):
+    return {CUSTOM_OBJ_NAME}("Custom-Object"), config.some_option
+"""
+
+define_sys_path_definition = """
+import sys
+import os
+
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+"""
+
+pipeline_definition = f"""
+from zenml.pipelines import pipeline
+
+@pipeline(enable_cache=False)
+def {PIPELINE_NAME}(
+        step_1,
+):
+    step_1()
+"""
+
+
+def test_pipeline_run_single_file(clean_repo, tmp_path) -> None:
+    """Test that zenml pipeline run works as expected when the pipeline, its
+    steps and materializers are all in the same file."""
+    runner = CliRunner()
+
+    os.chdir(str(tmp_path))
+    Repository.initialize()
+    clean_repo.activate_root()
+
+    # Create run.py file with steps, custom object, custom materializer and
+    #  pipeline definition
+    main_python_file = clean_repo.root / "run.py"
+    main_python_file.write_text(
+        "\n".join(
+            [
+                object_definition,
+                materializer_definition,
+                step_definition,
+                pipeline_definition,
+            ]
+        )
+    )
+
+    # Create run config
+    run_config = {
+        "name": PIPELINE_NAME,
+        "steps": {
+            "step_1": {
+                "source": {"name": STEP_NAME},
+                "parameters": {"some_option": 3},
+                "materializers": {"output_1": {"name": MATERIALIZER_NAME}},
+            }
+        },
+    }
+    config_path = str(clean_repo.root / "config.yaml")
+    yaml_utils.write_yaml(config_path, run_config)
+
+    # Run Pipeline
+    runner.invoke(pipeline, ["run", "run.py", "-c", "config.yaml"])
+
+    # Assert that the pipeline ran successfully
+    historic_pipeline = Repository().get_pipeline(pipeline_name=PIPELINE_NAME)
+
+    assert len(historic_pipeline.runs) == 1
+
+    assert historic_pipeline.runs[-1].status == ExecutionStatus.COMPLETED
+
+
+def test_pipeline_run_multifile(clean_repo, tmp_path) -> None:
+    """Test that zenml pipeline run works as expected when the pipeline, its
+    steps and materializers are all in the different files.
+
+    This test creates a project with the following structure.
+    |custom_obj_file
+    |   |--custom_obj_file.py
+    |materializer_file
+    |   |--materializer_file.py
+    |step_file
+    |   |--step_file.py
+    |config.yaml
+    |run.py
+    """ ""
+
+    STEP_FILE = "step_file"
+    MATERIALIZER_FILE = "materializer_file"
+    CUSTOM_OBJ_FILE = "custom_obj_file"
+
+    CliRunner()
+
+    os.chdir(str(tmp_path))
+    Repository.initialize()
+    clean_repo.activate_root()
+
+    # Write pipeline definition, add file to sys path to make sure the other
+    #  modules will be able to import from one another
+    new_pipeline_definition = (
```

Comment:
I have a question regarding this `sys.path` definition. Isn't that something that we explicitly don't want to have in our main pipeline files as it should work without that as well?

I'm guessing if we were to use `CliRunner.invoke` it might have been necessary, but when calling it using subprocess the entire thing should work without it IMO.

Reply:
Fixed this as discussed